public/pages/workflows/new_workflow/new_workflow.tsx

```diff
@@ -45,7 +47,7 @@ export function NewWorkflow(props: NewWorkflowProps) {
 
   // initial state
   useEffect(() => {
-    dispatch(getWorkflowPresets());
+    dispatch(getWorkflowPresets({ dataSourceId }));
```

Comment:
This call is actually only hitting server-side, and never hits any OpenSearch API, hence does not need any data source ID propagation. It is just to fetch the preset workflows we have stored under `server/resources/templates`. Can you remove here, in the reducer, and on server-side?